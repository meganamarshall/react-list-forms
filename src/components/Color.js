import React from 'react';
import PropTypes from 'prop-types';

const hexFromColor = color => {
  return `${color.toString(16).padStart(2, 0)}`.toUpperCase();
};

const hexFromRGB = (rgb) => {
  return `#${hexFromColor(rgb.red)}${hexFromColor(rgb.green)}${hexFromColor(rgb.blue)}`;
};

function Color({ name, rgb }) {
  const hex = hexFromRGB(rgb);

  const colorStyles = {
    backgroundColor: hex,
    width: '100px',
    height: '400px'
  };

  return (
    <dl style={colorStyles}>
      <dt>Name:</dt>
      <dd>{name}</dd>

      <dt>Hex:</dt>
      <dd>{hex}</dd>

      <dt>RGB:</dt>
      <dd>
        <p>Red - {rgb.red}</p>
        <p>Green - {rgb.green}</p>
        <p>Blue - {rgb.blue}</p>
      </dd>
    </dl>
  );
  
}
  
Color.propTypes = {
  name: PropTypes.string.isRequired,
  rgb: PropTypes.shape({
    red: PropTypes.number.isRequired,
    green: PropTypes.number.isRequired,
    blue: PropTypes.number.isRequired
  })
};

export default Color;
