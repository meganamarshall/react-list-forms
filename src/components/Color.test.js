import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color component', () => {
  it('renders the color', () => {
    const color = {
      name: 'blue',
      hex: 'ff0000'
    };
    const wrapper = shallow(<Color name={color.name} hex={color.hex} />);
    expect(wrapper).toMatchSnapshot();
  })
})
