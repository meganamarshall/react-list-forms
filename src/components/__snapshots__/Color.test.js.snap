// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component renders the color 1`] = `
<dl>
  <dt>
    Name:
  </dt>
  <dd>
    blue
    <div
      style={
        Object {
          "backgroundColor": "ff0000",
          "height": "100px",
          "width": "100px",
        }
      }
    />
  </dd>
  <dt>
    Hex:
  </dt>
  <dd>
    ff0000
  </dd>
  <dt>
    RGB:
  </dt>
  <dd>
    <p>
      Red - 
      240
    </p>
    <p>
      Green - 
      0
    </p>
    <p>
      Blue - 
      0
    </p>
  </dd>
</dl>
`;
